### ISSUE TITLE
ENDF read error for JEDNL radioactive decay data

### ISSUE THREAD

**Issue by evan012345** [openmc-dev/openmc#1221]
Hi all,
I came across a bug or issue reading ENDF files into the OpenMC radioactive decay class (openmc.data.Decay). To be fair, it has been working great for ENDF/B-VIII.0 and JEFF-3.3 ENDF formatted nuclear data, but crashes when reading JENDL data. I suspect that this might be an issue with the ENDF formatted data in JENDL rather than OpenMC, and could be somewhat related to this past [issue](https://github.com/openmc-dev/openmc/issues/1161).

Anyway... A number of JENDL-FPD-2011 isotope files are successfully read until it reaches Co-66 (027_Co_066.jfpd11). See error message below. If I remove Co-66, the code simply crashes at a different isotope, with the same error.

Is there a known hack or easy workaround? I really would like to read the JENDL data in OpenMC.

Thanks,
Evan

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-b73d7c14ab61> in <module>()
      9         # open endf file using openmc
     10         print(entry.path)
---> 11         endf_data = openmc.data.Decay(entry.path)
     12 
     13         # get nuclide info

~/anaconda3/lib/python3.7/site-packages/openmc/data/decay.py in __init__(self, ev_or_filename)
    427                             di['positron_intensity'] = ufloat(*values[4:6])
    428                         elif spectrum['type'] == 'gamma':
--> 429                             di['internal_pair'] = ufloat(*values[4:6])
    430                             if len(values) >= 8:
    431                                 di['total_internal_conversion'] = ufloat(*values[6:8])

TypeError: ufloat() missing 1 required positional argument: 'nominal_value'

**Comment by paulromano** [comment]
Hi @evan012345. If you want a quick hack to get the `Decay` class working with that file, you can make the following change:
```diff
diff --git a/openmc/data/decay.py b/openmc/data/decay.py
index bbb059f4f..c165c32bc 100644
--- a/openmc/data/decay.py
+++ b/openmc/data/decay.py
@@ -423,7 +424,8 @@ class Decay(EqualityMixin):
                         if spectrum['type'] == 'ec/beta+':
                             di['positron_intensity'] = ufloat(*values[4:6])
                         elif spectrum['type'] == 'gamma':
-                            di['internal_pair'] = ufloat(*values[4:6])
+                            if len(values) >= 6:
+                                di['internal_pair'] = ufloat(*values[4:6])
                             if len(values) >= 8:
                                 di['total_internal_conversion'] = ufloat(*values[6:8])
                             if len(values) == 12:
```

It's not clear to me whether this would be considered a bug in OpenMC or a bug in the file itself. It currently chokes when trying to read a discrete gamma spectrum. The ENDF-6 manual has this to say about gamma spectra in decay files:
> For gamma ray emission (STYP=0.0), no other information is required if X-ray, Auger electron, conversion electron, and pair formation intensities have not been calculated for these transitions. In this case NT=6.

In this file, a gamma spectrum appears with NT=4 but OpenMC assumes that NT >= 6 and tries to use values accordingly.

I tried all the other files in JENDL/FPD-2011 and noticed that many of them do not have a MF=8, MT=457 section (evidently indicating that the nuclide is stable). To be clear, the evaluator _is_ supposed to put in this section even for a stable nuclide since there is a flag (NST) that indicates stability. Just be aware that OpenMC will choke on these files too, but I think that is a more clear cut case of the evaluation being "wrong".

Anyway, hope you get it working for your use case, and happy to see someone using `openmc.data` (which is a sometimes underappreciated part of the project)!

**Comment by paulromano** [comment]
Fixed with the merging of #1271. I've confirmed that the original JENDL FPD file (Co66) can now be read successfully. However, many other JENDL FPD files will fail due to the lack of the MF=8, MT=457 section.
